application/api/auth.2/error.js

```diff
@@ -0,0 +1,16 @@
+({
+  AuthError: class AuthError extends DomainError {
+    static ALREADY_EXISTS() {
+      return new AuthError(409, { message: 'User already exists' });
+    }
+    static INVALID_CREDENTIALS() {
```

Comment:
Why not create separate AuthError classes instead?
```
class AuthError extends DomainError { code: 401 }
class AuthExistsError extends AuthError { code: 409 }
class AuthInvalidCredentialsError extends AuthError { code 400 }
...
```
it should be simpler to manage and use.